images/nix/default.nix

```diff
@@ -42,14 +44,18 @@ let
 
       # make sure /tmp exists
       mkdir -m 1777 tmp
+
+      # need a HOME
+      mkdir -vp root
     '';
 
     config = {
       Cmd = [ "/bin/bash" ];
       Env = [
         "ENV=/etc/profile.d/nix.sh"
+        "BASH_ENV=/etc/profile.d/nix.sh"
         "NIX_BUILD_SHELL=/bin/bash"
-        "NIX_PATH=nixpkgs=${toString <nixpkgs>}"
+        "NIX_PATH=nixpkgs=${pkgs.path}"
```

Comment:
Actually the intent was to get something like `NIX_PATH=nixpkgs=channel:nixos-20.03`. Users should use pinned nixpkgs and this is just a fallback.

Reply:
makes sense to me.

Reply:
ok.  NIX_PATH here confused me because it existed so i tried to use it.
Would it be possible to either document the intended usage or set it to something more explicit?

Reply:
do you mind sending another PR to restore the channel name? I think it will require to change the CI config as well

Reply:
Actually, let's make the failure more explicit. That way it's obvious when people forget to pin their nixpkgs: https://github.com/nix-community/docker-nixpkgs/pull/9